transport_nantes/mobilito/views.py

```diff
@@ -67,16 +67,14 @@ def get(self, request, *args, **kwargs):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        this_page = kwargs.get('tutorial_page', 'presentation')
-        if this_page not in self.all_tutorial_pages:
+        self.this_page = kwargs.get('tutorial_page', 'presentation')
+        if self.this_page not in self.all_tutorial_pages:
             # Trying to access a tutorial page that doesn't exist.
             raise Http404()
-        context['active_page'] = this_page
-        self.template_name = f'mobilito/tutorial_{this_page}.html'
+        context['active_page'] = self.this_page
```

Comment:
Why are you putting `this_page` on the object (on `self`) instead of passing it as an argument to `get_seen_pages_from_cookie()`?

In general, passing arguments makes dependencies clear, whereas assigning to global state (or to larger than necessary state) makes code harder to read, because it's hard to know where things come from, what came from what.

Reply:
Moved to args, removed from self